### PR TITLE
fix: limit max size

### LINKED ISSUE #11: Define max filesize of accepted _redirects file
This library should have a sane default around maximum allowed size of the _redirects file, to remove DoS vector at places where arbitrary CID root can be loaded (e.g., subdomain gateways)

There is some prior art in https://gitlab.com/gitlab-org/gitlab-pages/-/issues/472, according to their use case a very big website with ton of redirect rules should still be no more than 64 KiB. 

I propose we use this as a hard limit for now and error `Parse` when more than 64 KiB of rules is read. 
This can be a lazy-check performed as we go, this is just a precaution at the library level, in case someone prepares fake dag-pb node with spoofed size value to try to work around any size check done in Kubo. 

If this sounds fine, please:
- add test that confirms library errors when file is bigger than the limit.
- add a paragraph about the maximum size to https://github.com/ipfs/specs/pull/290 so alternative implementations do the same.

### PR DESCRIPTION
**Pull request by justindotpub**
Closes #11

Will need to rebase after https://github.com/ipfs/go-ipfs-redirects-file/pull/14 is merged.

### CODE REVIEW

**Review by lidel — APPROVED** [review]

File: redirects.go [hunk 1]
```diff
@@ -94,8 +98,16 @@ func Must(v []Rule, err error) []Rule {
...
+	b, err := bufio.NewReaderSize(r, maxFileSizeInBytes+1).Peek(maxFileSizeInBytes + 1)
+	if err != nil && err != io.EOF {
+		return nil, err
+	}
+	if len(b) > maxFileSizeInBytes {
+		return nil, fmt.Errorf("redirects file size cannot exceed %d bytes", maxFileSizeInBytes)
+	}
 
+	s := bufio.NewScanner(bytes.NewReader(b))
 	for s.Scan() {
```
Comment (lidel) [inline redirects.go:111]: @Jorropo any room for improvement here? I guess we need to read entire thing anyway, so does not matter much, but asking just to be sure i am not missing something obvious.

Comment (Jorropo) [reply redirects.go:111]: I don't see why you use `bufio.Reader` with a `.Peek` this large, it's a complex beast inside.

You can just do `io.ReadAll(&io.LimitedReader{r, maxFileSizeInBytes+1})`.

Comment (Jorropo) [reply redirects.go:111]: You can even feed the `LimitedReader` directly to the scanner if you want.

Comment (Jorropo) [reply redirects.go:111]: You could also match rules while reading instead of returning a list of rules, allowing for a fully streaming implementation.

Comment (lidel) [reply redirects.go:111]: Thanks, I've switched to simpler code with LimitedReader and streaming in https://github.com/ipfs/go-ipfs-redirects-file/pull/18

**Pull request merged**